### PR TITLE
feat(templates,docs): add tutorial to templates

### PR DESCRIPTION
**Pull request by brookslybrand**
Adding @ryanflorence's [remix-docs-template](https://github.com/ryanflorence/remix-tutorial-template) to the templates directory. I could be remembering wrong, but I believe Ryan mentioned that this should have been in this repo originally.

Beyond just copying and pasting, I made 3 other updates:
1. Use whatever the latest version of Remix is by default
2. Added `--manual` as that seems to fix a race condition [many users are experiencing](https://github.com/remix-run/remix/issues/7466)
3. Updated the `create-remix` command in `tutorial.md`

I thought running the following would allow you to test this, but I'm getting a 404 from GitHub and I'm not really sure why

```sh
npx create-remix@latest --template https://github.com/remix-run/remix/tree/brooks/remix-tutorial-template/templates/remix-tutorial
```

### CODE REVIEW

**Review by brophdawg11 — APPROVED** [review]

**Comment by brookslybrand** [comment]
> > I thought running the following would allow you to test this, but I'm getting a 404 from GitHub and I'm not really sure why
> 
> Hm, yeah I would expect that to work as well. Maybe it has something to do with the slash in the branch name. If you want to open a separate issue and assign to me I can dig into that for the CLI

[Done!](https://github.com/remix-run/remix/issues/8151)

I used a file path to test locally, so that works. But yeah, seems to be something funky about the branch name. I appreciate you looking into it!

File: templates/remix-tutorial/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/** @type {import('eslint').Linter.Config} */
+module.exports = {
+  extends: ["@remix-run/eslint-config", "@remix-run/eslint-config/node"],
```
[the remark this answers is not in the document]
Comment (brookslybrand) [reply templates/remix-tutorial/.eslintrc.js]: Or not, seems like the linting isn't passing in the repo 🙃 

I'll keep goofing with it

[the remark this answers is not in the document]
Comment (brookslybrand) [reply templates/remix-tutorial/.eslintrc.js]: Alright @brophdawg11, fixed those errors. Let me know whether or not that seems like a fine setup

File: templates/remix-tutorial/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+          { name: "Link", linkAttribute: "to" },
+          { name: "NavLink", linkAttribute: "to" },
+        ],
+      },
+      rules: {
+        "react/jsx-no-leaked-render": [
+          "warn",
+          { validStrategies: ["ternary"] },
+        ],
+      },
```
Comment (brophdawg11) [inline templates/remix-tutorial/.eslintrc.js:46]: For the purposes of a tutorial we can probably trim this also and just rely on the "recommended" rulesets

File: templates/remix-tutorial/.eslintrc.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (brophdawg11) [reply templates/remix-tutorial/.eslintrc.js]: Yeah that looks good to me with one small additional potential removal.  Our goal is provide the "best practice" rules (as defined by a "recommended" config) for any core tech used in the Remix app (which will differ by template/stack/tutorial etc).  So React + TypeScript + Node configs make sense here 👍

**Pull request merged**